### DISCUSSION TITLE
Collection item fields not instantly exposed when API request is filtered by a roles custom permission

### DISCUSSION THREAD

**Discussion by domdus** [directus/directus#14948] [Archive]
### Describe the Bug

Hope it's not a local issue with my env, but I can reproduce it and will try to explain the potential bug.

We have content collections e.g. articles, videos, ... and the content relates to customers ("services" M2M).
This way we assign content to "services" resulting in the designated display for various portals & apps.

I use a roles **custom permissions** for content collections to only expose content items assigned to the client.

### Here comes the issue:
Whenever I do a fresh setup of a user and role + custom permission on content collection - On first API request, the collection items are only presented with their id's (primary keys). I specified `fields=*`:
<img width="386" alt="image" src="https://user-images.githubusercontent.com/67951828/183395650-6218d2cc-c01a-4486-8afa-071ddbd136b9.png">

Then I need to change the custom permission once to `all access`:
<img width="394" alt="image" src="https://user-images.githubusercontent.com/67951828/183407591-6fd92cc9-4378-465e-b69d-23b67fec019a.png">

And finally the items appear with their fields:
<img width="367" alt="image" src="https://user-images.githubusercontent.com/67951828/183396776-0b277f71-6032-4dd2-8a50-2c60684193e4.png">

After that I can put back in the custom-permission to receive only items allowed for the service.

The custom permission I set is:
```
{
	"_and": [
		{
			"service": {
				"services_id": {
					"key": {
						"_eq": "test"
					}
				}
			}
		}
	]
}
```

### Note:
It is not cache or something that resolves over time! Directus Cache is disabled, also app restart does not fix the behavior until the permission is removed once.

### To Reproduce

(Simplified)

1. create a collection `test` with some fields [title, description]
2. add test items to collection with title `item1`, `item2`, `item3`
3. create a user `test` and create static token for API access
4. create a role `test`
5. assign user `test` to role `test`
6. give custom permission for `read` of collection `test`
   `{
	"_and": [
		{
			"title": {
				"_eq": "item1"
			}
		}
	]
   }`
<img width="428" alt="image" src="https://user-images.githubusercontent.com/67951828/183400426-d87e176a-ab8b-4c0e-987c-19074345738a.png">

7. Request API at /items/test with the user created (static token) and see item missing fields. Try to add query ?fields=*,title, ...whatever:
<img width="311" alt="image" src="https://user-images.githubusercontent.com/67951828/183400854-ff15a64e-5466-4680-980d-758ced762560.png">

The fields do not appear until you remove the custom permission once.

8. Remove custom permission, grant `all access` for `read` and request API again:
<img width="351" alt="image" src="https://user-images.githubusercontent.com/67951828/183401828-167d1c75-20c0-426d-aa86-18befa5a3ef2.png">

9. After fields appear, put back in custom permission to see `item1` correctly returned and fields are avaiable too.
<img width="342" alt="image" src="https://user-images.githubusercontent.com/67951828/183402541-e8f541c9-1f0c-4aa5-8636-1ac35e96fc04.png">

### What version of Directus are you using?

9.14.5

### What version of Node.js are you using?

16.14.0

### What database are you using?

Postgres 13

### What browser are you using?

Chrome

### How are you deploying Directus?

PaaS

**Comment by licitdev** [comment]
@domdus As the default permission is `No Access`, no field is allowed. Thus when setting a custom permission, you will have to select which field is allowed in the `Field Permissions` tab.

![](https://user-images.githubusercontent.com/67951828/183400426-d87e176a-ab8b-4c0e-987c-19074345738a.png)

When you set to `All Access`, all fields are allowed. Thus when you were to set `Use Custom` after, all fields are allowed.

The status indicator on the `Field Permissions` tab will be highlighted if there is enabled field(s).

**Reply by domdus** [reply]
Ahh okay got it. Now I noticed how fields all set to allow after set `All Access` once. Many thx!
